### PR TITLE
Create GCP specific assets: infrastructure, cred secrets, & dns

### PR DESCRIPTION
**Pull request by abhinavdahiya**
JIRA: [CORS-1149](https://jira.coreos.com/browse/CORS-1149)

This PR writes manifests for several assets required by GCP: infrastructure, cloud cred secrets, and dns.

* Add GetSession to load creds for GCP
It reads the service account JSON content or file from following locations in order:
// env GOOGLE_CREDENTIALS,
// env GOOGLE_CLOUD_KEYFILE_JSON,
// env GCLOUD_KEYFILE_JSON,
// file ~/.gcp/osServiceAccount.json, and
// gcloud cli defaults

If none of the above are succesful in loading the credentials, it prompts the user to provide the location of the service account json or paste the content to the prompt.

The creds are initialized for `https://www.googleapis.com/auth/cloud-platform` scope.

* Create DNS manifest for GCP.
Add logic to write cluster DNS manifest for GCP. Instantiates a client to the GCP API, which retrieves a zone that matches the provided base domain.
Zone's name is enough to get the zone for operators in a project
https://cloud.google.com/dns/docs/reference/v1/managedZones/get#parameters

* Create cloud credentials secret for GCP.
Add GCP service account JSON to cloud cred secret.
For GCP the `cloud-cred-secret` is created with name `gcp-credentials` and stores the service account key in JSON format in the key `service_account.json`

Most of the changes are from https://github.com/openshift/installer/pull/1927 with fixes around the GCP API access with credentials.

Closes https://github.com/openshift/installer/pull/1927

/cc @patrickdillon @jstuever @csrwng

### CODE REVIEW

File: pkg/asset/installconfig/gcp/dns.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+	"github.com/pkg/errors"
+	dns "google.golang.org/api/dns/v1"
+	"google.golang.org/api/option"
+)
+
+// GetPublicZone returns a DNS managed zone from the provided project which matches the baseDomain
+// If multiple zones match the basedomain, it uses the last public zone in the list as provided by the GCP API.
+func GetPublicZone(ctx context.Context, project, baseDomain string) (managedZone *dns.ManagedZone, err error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
```
Comment (csrwng) [inline pkg/asset/installconfig/gcp/dns.go]: you use the same context with a timeout of 1 minute to page through all managed dns zones. Is there any concern with high latency connections and many zones?

Comment (abhinavdahiya) [reply pkg/asset/installconfig/gcp/dns.go]: we are filtering on DNS Zones matching on `basedomain`
https://github.com/openshift/installer/pull/1944/commits/2b7bd18d8efcc2b256eaa100dc42d40b47f45932#diff-4475e438b327663f0625c6bfef1d2dd4R33

And there shouldn't be a whole lot of those...

**Pull request merged**